packages/react-query/src/useQueryOptions.ts

```diff
@@ -0,0 +1,31 @@
+import type {DefaultedQueryObserverOptions, QueryClient, QueryKey, QueryObserverOptions} from "@tanstack/query-core";
+import {useQueryClient} from "./QueryClientProvider";
+import type React from "react";
+import type { DependencyList } from "react";
+import {useMemo} from "react";
+
+export function useQueryOptions<
```

Comment:
How are these two APIs used ? The PR only contains declarations and exports.

Aren't useQueryOptions a combination of what the user gives and default options? in this case, how this hook is adding any value ?

Reply:
sorry this was in progress at the time, based on the above feedback. I have updated it now. It ties into a larger picture here.